Rules/UseOutputTypeCorrectly.cs

```diff
@@ -0,0 +1,179 @@
+            {
+                return AstVisitAction.Continue;
+            }
+
+            HashSet<string> outputTypes = new HashSet<string>();
+
+            foreach (AttributeAst attrAst in funcAst.Body.ParamBlock.Attributes)
+            {
+                if (attrAst.TypeName != null && attrAst.TypeName.GetReflectionType() == typeof(OutputTypeAttribute)
+                    && attrAst.PositionalArguments != null)
+                {
+                    foreach (ExpressionAst expAst in attrAst.PositionalArguments)
+                    {
+                        if (expAst is StringConstantExpressionAst)
+                        {
+                            Type type = Type.GetType((expAst as StringConstantExpressionAst).Value);
+                            if (type != null)
+                            {
+                                outputTypes.Add(type.FullName);
+                            }
+                        }
+                        else
+                        {
+                            TypeExpressionAst typeAst = expAst as TypeExpressionAst;
+                            if (typeAst != null && typeAst.TypeName != null)
+                            {
+                                if (typeAst.TypeName.GetReflectionType() != null)
+                                {
+                                    outputTypes.Add(typeAst.TypeName.GetReflectionType().FullName);
+                                }
+                                else
+                                {
+                                    outputTypes.Add(typeAst.TypeName.FullName);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            List<Tuple<string, StatementAst>> returnTypes = FindPipelineOutput.OutputTypes(funcAst, _classes);
+
+            foreach (Tuple<string, StatementAst> returnType in returnTypes)
+            {
+                string typeName = returnType.Item1;
+
+                if (String.IsNullOrEmpty(typeName)
+                    || String.Equals(typeof(Unreached).FullName, typeName, StringComparison.OrdinalIgnoreCase)
+                    || String.Equals(typeof(Undetermined).FullName, typeName, StringComparison.OrdinalIgnoreCase)
+                    || String.Equals(typeof(object).FullName, typeName, StringComparison.OrdinalIgnoreCase)
+                    || outputTypes.Contains(typeName, StringComparer.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+                else
+                {
+                    DiagnosticRecords.Add(new DiagnosticRecord(string.Format(CultureInfo.CurrentCulture, Strings.UseOutputTypeCorrectlyError,
+                        funcAst.Name, typeName), returnType.Item2.Extent, GetName(), DiagnosticSeverity.Warning, fileName));
```

Comment:
I think the severity must be information. Expecting users to have outputtype attribute for every single function is hard. Opinions?